code/modules/atmospherics/machinery/components/tank.dm

```diff
@@ -0,0 +1,521 @@
+	air_contents.set_volume(volume)
+	RefreshPressureLimit()
+
+	if(gas_type)
+		FillToPressure(gas_type)
+
+	setPipingLayer(piping_layer)
+	QUEUE_SMOOTH(src)
+	QUEUE_SMOOTH_NEIGHBORS(src)
+
+	return INITIALIZE_HINT_LATELOAD
+
+// We late initialize here so all stationary tanks have time to set up their
+// initial gas mixes and signal registrations.
+/obj/machinery/atmospherics/components/tank/LateInitialize()
+	. = ..()
+	GetMergeGroup(merger_id, merger_typecache)
+
+/obj/machinery/atmospherics/components/tank/Destroy()
+	QUEUE_SMOOTH_NEIGHBORS(src)
+	return ..()
+
+/obj/machinery/atmospherics/components/tank/examine(mob/user, thats)
+	. = ..()
+	var/wrench_hint = EXAMINE_HINT("wrench")
+	if(!initialize_directions)
+		. += "<span class='notice'>A pipe port can be opened with a [wrench_hint].</span>"
+	else
+		. += "<span class='notice'>The pipe port can be moved or closed with a [wrench_hint].</span>"
+
+/obj/machinery/atmospherics/components/tank/set_custom_materials(list/materials, multiplier)
+	. = ..()
+	RefreshPressureLimit()
+
+/// Recalculates pressure based on the current max integrity compared to original
+/obj/machinery/atmospherics/components/tank/proc/RefreshPressureLimit()
+	var/max_pressure_multiplier = max_integrity / initial(max_integrity)
+	max_pressure = max_pressure_multiplier * initial(max_pressure)
+
+/// Fills the tank to the maximum safe pressure.
+/// Safety margin is a multiplier for the cap for the purpose of this proc so it doesn't have to be filled completely.
+/obj/machinery/atmospherics/components/tank/proc/FillToPressure(gastype, safety_margin=0.5)
+	var/pressure_limit = max_pressure * safety_margin
+
+	var/moles_to_add = (pressure_limit * air_contents.return_volume()) / (R_IDEAL_GAS_EQUATION * air_contents.return_temperature())
+	air_contents.adjust_moles(gastype, moles_to_add)
+	air_contents.archive()
+	update_parents()
+
+/obj/machinery/atmospherics/components/tank/process_atmos()
+	if(air_contents.react(src))
+		update_parents()
+
+	if(air_contents.return_pressure() > max_pressure)
+		take_damage(0.1, BRUTE, sound_effect=FALSE)
+		if(prob(40))
+			playsound(src, pick(breaking_sounds), 30, vary=TRUE)
+
+	RefreshWindow()
+
+///////////////////////////////////////////////////////////////////
+// Pipenet stuff
+
+/obj/machinery/atmospherics/components/tank/returnAirsForReconcilation(datum/pipeline/requester)
+	. = ..()
+	if(!air_contents)
+		return
+	. += air_contents
+
+/obj/machinery/atmospherics/components/tank/returnPipenetsForReconcilation(datum/pipeline/requester)
+	. = ..()
+	var/datum/merger/merge_group = GetMergeGroup(merger_id, merger_typecache)
+	for(var/obj/machinery/atmospherics/components/tank/tank as anything in merge_group.members)
+		. += tank.parents
+
+/obj/machinery/atmospherics/components/tank/proc/ToggleSidePort(new_dir)
+	if(initialize_directions & new_dir)
+		initialize_directions &= ~new_dir
+	else
+		initialize_directions |= new_dir
+
+	for(var/i in 1 to length(nodes))
```

Comment:
i think single letter variables that are just interators are fine left as is, incase i want to change that here it is though